### ISSUE TITLE
export onnx

### ISSUE THREAD

**Issue by FlowEternal** [aliyun/conditional-lane-detection#13]
I tried this model on my own video, hoping it can somehow improve the detection performance on corner cases like Y branching, but it fail to work as what I expected when using pretrained curvelanes model. 

One vital drawback of this model is that perhaps it is extremely hard to be deployed on embedding devices, because there exist conditional weight/bias and conditional branching which depending on the input image on the fly.  

So what is the suggestion to export it to onnx with dynamically weight and structure.  Considering the fiexed nature of onnx inference framwwork like onnx runtime, really hard to do that!

**Comment by lucky-xu-1994** [comment]
> I tried this model on my own video, hoping it can somehow improve the detection performance on corner cases like Y branching, but it fail to work as what I expected when using pretrained curvelanes model.
> 
> One vital drawback of this model is that perhaps it is extremely hard to be deployed on embedding devices, because there exist conditional weight/bias and conditional branching which depending on the input image on the fly.
> 
> So what is the suggestion to export it to onnx with dynamically weight and structure. Considering the fiexed nature of onnx inference framwwork like onnx runtime, really hard to do that!
yes, it's really hard to export to onnx. I have tried a lot, but failed for some Variables would be frozen to constants.

**Comment by lucky-xu-1994** [comment]
@FlowEternal 
 yes, it's really hard to export to onnx. I have tried a lot, but failed for some Variables would be frozen to constants.
so, if you have any progress,please share here! thank you!

**Comment by FlowEternal** [comment]
> @FlowEternal 
>  yes, it's really hard to export to onnx. I have tried a lot, but failed for some Variables would be frozen to constants.
> so, if you have any progress,please share here! thank

no, I will not waste my time doing this because it doesn't fit into our project.  It has inherent disadvantage of not being able to detect nearly horizontal lane or extremely curved lane, because it constrain the direction of lanes like other similar models. It will fail when dealing with surround view perception like what Tesla is doing

**Comment by lucky-xu-1994** [comment]
> > @FlowEternal
> > yes, it's really hard to export to onnx. I have tried a lot, but failed for some Variables would be frozen to constants.
> > so, if you have any progress,please share here! thank
> 
> no, I will not waste my time doing this because it doesn't fit into our project. It has inherent disadvantage of not being able to detect nearly horizontal lane or extremely curved lane, because it constrain the direction of lanes like other similar models. It will fail when dealing with surround view perception like what Tesla is doing

Anyway, thank you for your reply！

**Comment by sigma-alpha-beta** [comment]
> 
> 
> > @FlowEternal
> > yes, it's really hard to export to onnx. I have tried a lot, but failed for some Variables would be frozen to constants.
> > so, if you have any progress,please share here! thank
> 
> no, I will not waste my time doing this because it doesn't fit into our project. It has inherent disadvantage of not being able to detect nearly horizontal lane or extremely curved lane, because it constrain the direction of lanes like other similar models. It will fail when dealing with surround view perception like what Tesla is doing

hello，Do you have any advice on how to deal with near-horizontal lane? or  any other models can detect nearly horizontal lane. I tried add a new branch to handle horizontal lane on other model, but it can ony handle a specific type of lanes, for example a stop lane. thanks

**Comment by 18112330636** [comment]
This can be deployed. There is a precedent before DCN variability convolution
Its convolution layer output also goes to the convolution kernel
Just write a tensorrt plug-in
Onnx only stores the data flow of model structure
It has nothing to do with values

**Comment by abhigoku10** [comment]
@18112330636  can you share the conversion scripts or the steps followed to convert into onnx

**Comment by LeungWaiHo** [comment]
if I only use the tusimple version model, will it have the same problem?
